include/secp256k1.h

```diff
@@ -370,6 +371,21 @@ SECP256K1_API int secp256k1_ec_pubkey_serialize(
     unsigned int flags
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
+/** Compare two public keys using lexicographic (of compressed serialization) sort
+ *
+ *  Returns: <0 if the first public key is less than the second
+ *           >0 if the first public key is greater than the second
+ *           0 otherwise
+ *  Args: ctx:      a secp256k1 context object.
+ *  In:   pubkey1:  first public key to compare
+ *        pubkey2:  second public key to compare
```

Comment:
This is implicit as per the discussion in https://github.com/bitcoin-core/secp256k1/pull/783#issuecomment-695082220

Reply:
Okay, yes. Strictly speaking, we first need to add the line `Unless explicitly stated all pointer arguments must not be NULL.` in the header file (as done in #783) because currently we promise that the `illegal_callback` is only called for violations explicitly mentioned in the header.

Reply:
added that line in #926